### PR TITLE
EOS-26683: Accept SB Filters - Components, binlogs, stacktrace etc

### PR DESCRIPTION
**Pull request by gargpalak048**
Signed-off-by: Palak Garg <palak.garg@seagate.com>

# Problem Statement
- Accept SB Filters - Components, binlogs, coredumps, stacktrace etc

# Design
-  For Bug describe the fix here. 
-  For Feature, Post the link to the solution page on the confluence CORTX Foundation Library 

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [x] Changes done to WIKI / Confluence page
https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/749076684/Node+level+Support+Bundle+Collection

### CODE REVIEW

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 1]
```diff
@@ -49,6 +49,33 @@ def _generate_bundle_id():
...
+            for comp_idx in range(0, num_components):
+                services = cortx_config_store.get(
+                        f'node>{node_id}>components[{comp_idx}]>services')
+                service = 'all' if services is None else ','.join(services)
+                comp_name = components[comp_idx]['name']
+                components_list.append(comp_name)
+                service_per_comp[comp_name] = service
+        else:
+            comp_list = components.lower().split(',')
+            for comp in comp_list:
```
Comment (cdeshmukh) [inline py-utils/src/utils/support_framework/support_bundle.py:69]: add expected example input here, so its easy to relate to the code

File: py-utils/src/utils/support_framework/support_bundle.py [hunk 2]
```diff
@@ -49,6 +49,33 @@ def _generate_bundle_id():
...
+                if '|' in comp:
+                    comp_name = comp.split(':')[0]
+                    service_per_comp[comp_name] = comp.split(':')[1].replace('|', ',')
+                    components_list.append(comp_name)
+                else:
+                    components_list.append(comp)
+                    service_per_comp[comp] = 'all'
```
Comment (cdeshmukh) [inline py-utils/src/utils/support_framework/support_bundle.py:76]: here there is no scope of services coming in argument, specially when there is one component name ?

File: py-utils/src/utils/support_framework/support_bundle.py (hunk 1 above)
Comment (gargpalak048) [reply py-utils/src/utils/support_framework/support_bundle.py:72]: Added.

File: py-utils/src/utils/support_framework/support_bundle.py (hunk 2 above)
Comment (gargpalak048) [reply py-utils/src/utils/support_framework/support_bundle.py:79]: Expected str will be something like -> "S3:Authserver|BGProducer,CSM,UTILS"
if ":" & "|" not found for component, then default 'all' will be set

**Review by cdeshmukh — APPROVED** [review]

**Comment by rohit-k-dwivedi** [comment]
@sachinpunadikar , PR looks good to me

**Review by rohit-k-dwivedi — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**